841 "Fjern/velg alle" resultater fra søk i filter

robinzaal opened the pull request:
This closes #841 and closes #853

Added functionality to check/uncheck list items from a searched result. Cleaned up markup and updated to new syntax.

robinzaal committed:
Update with new syntax

robinzaal committed:
Optimize rendering logic

robinzaal committed:
Move sections of code out of single-use `ng-template`'s to their respective places in the markup.

robinzaal committed:
Correctly conditionally render buttons and handle user click

proand commented on projects/fhi-angular-components/src/lib/fhi-tree-view-selection/fhi-tree-view-selection.component.ts:
Hmm... seems like `updateResultListHeighWhileLoading()` has stopped working.

proand replied:
@robinzaal My comment from Jun 23 still seems valid...

proand reviewed (changes requested):
Solution is becoming a bit sluggish with many items (testing with ca. 2500 items). Not sure if it's good enough or not. Maybe @VillereV can take a look at this as well?

PS. Here is a branch with a large dataset: https://github.com/folkehelseinstituttet/Fhi.Frontend.Demo/tree/841-select-searched--with-large-dataset

Commit:
Add large dataset

robinzaal committed:
Reintroduce `#resultListWrapper` and accompanying wrapper div

robinzaal committed:
Optimized selection of items

robinzaal committed:
Can select subset of items again

robinzaal committed:
Can select filtered subset (user search)

robinzaal committed:
Remove unused parameter and fix some typos.

proand commented:
Performance seems much better now 👍

But there is a bug when single selection: "Velg alle" doesn't change to "Fjern alle" unless you click on "Velg alle". See img below.

<img width="503" height="303" alt="image" src="https://github.com/user-attachments/assets/a637bb37-7bea-4c37-b21f-74b3414119cb" />

@robinzaal Merge  https://github.com/folkehelseinstituttet/Fhi.Frontend.Demo/tree/841-select-searched--with-large-dataset into your branch in this PR when your done working on the large dataset.

robinzaal committed:
Refactor `toggleChecked` to use item object directly

Updated the `toggleChecked` method to accept and process the item object instead of just its ID. This ensures we deal with the correct list of items (filtered vs non-filtered).

robinzaal committed:
Change back to demo dataset

robinzaal committed:
Merge branch 'dev' into 841-select-searched

proand commented:
> Performance seems much better now 👍
> 
> But there is a bug when single selection: "Velg alle" doesn't change to "Fjern alle" unless you click on "Velg alle". See img below.
> 

@robinzaal  My bad, read the issue more closely, and saw that showing "Fjern alle" only when all checkboxes are selected is the intended behavior.

robinzaal committed:
Update changelog
